.github/workflows/e2e.yaml

```diff
@@ -50,10 +50,11 @@ jobs:
         - name: Build e2e
           run: |
             cd e2e
-            test_dirs="$(ls -A tests)"
-            for td in $test_dirs
+            find ./tests -type d | while IFS= read -r dir
```

Comment:
finds all folders in `tests/` and if they contain a `.go` file, compiles the test binary

Reply:
So just so I understand this pipe - IFS = input field separator, and its separating on blank/space(?) reading from stdin into var `dir` which we then use inside the loop

Reply:
typically it is used with some separator value if required, setting it to nothing `IFS=` means it basically won't do any further splitting of the file paths (allowing e.g a path with spaces if one arises). A ref: https://unix.stackexchange.com/a/676870